include/rapidcheck/detail/Utility.h

```diff
@@ -95,7 +96,16 @@ inline uint64_t avalanche(uint64_t x) {
 /// and the rest set to 0.
 template <typename T>
 constexpr T bitMask(int nbits) {
-  return ~((~static_cast<T>(0) - 1) << static_cast<T>(nbits - 1));
+  using UT = typename std::make_unsigned<T>::type;
+  // There are two pieces of undefined behavior we're avoiding here,
+  //   1. Shifting past the width of a type (ex `<< 32` against an `int32_t`)
+  //   2. Shifting a negative operand (which `~0` is for all signed types)
+  // First we branch to avoid shifting the past the width of the type, then
+  // (assuming we are shifting, and aren't just returning `~0`) we cast `~0`
```

Comment:
We're actually casting `0` to an unsigned type, then inverting it and _then_ shifting it.